File: java/syntax/1.LogComments.java
// Log
public class LogComments {
    public static void main(String[] args) {
        // Hello world
        System.out.println("Hello world"); // Hello world
        System.out.println("Hello " + 23); // Hello 23

        // Inline string substitution
        System.out.printf("Hello %d%n", 23); // Hello 23

        // String interpolation
        String abc = String.format("Hello my age is %d", 10);
        System.out.println(abc);

        // Comments
        // This is a single line comment
        System.out.println("Hello, World!");

        // This is a multiline comment
        /*
        Multiline comment
        */

        // String concatenation
        String name = "John";
        int age = 25;
        System.out.println("Hello, " + name + "! You are " + age + " years old.");

        // Template strings (using String.format)
        String template = String.format("Hello, %s! You are %d years old.", name, age);
        System.out.println(template);
    }
} 

File: 01-syntax/python/1.log-comments.py
# Log
print("Hello world")  # Hello world
print("Hello", 23)  # Hello 23

# Inline string substitution
print(f"Hello {23}")  # Hello 23

# String interpolation
abc = f"Hello my age is {10}"
print(abc)

# Comments
# This is a single line comment
print("Hello, World!")

# This is a multiline comment
"""
Multiline comment
"""

# String concatenation
print("Hello, " + name + "! You are " + str(age) + " years old.")

# Template strings
from string import Template

template = Template("Hello, $name! You are $age years old.")
print(template.substitute(name=name, age=age))
